integration/forks/test_helpers

```diff
@@ -1,75 +1,72 @@
 #!/bin/bash
 
+source ../common
+
 build_images() {
   docker build -t geth ./geth
-	docker build -t smartcontract/chainlink ../../
+  docker build -t smartcontract/chainlink ../../
 }
 
 start_network() {
-	echo "Starting Network. See logs/{chain_[12],geth}.log for logs"
-	( docker-compose up > /dev/null ) &
+  echo "Starting Network. See logs/{chain_[12],geth}.log for logs"
+  ( docker-compose up > /dev/null ) &
 }
 
 tear_down() {
-	docker-compose down
-	# Remove CL database
-	sudo rm -rf tmp
+  docker-compose down
+  # Remove CL database
+  sudo rm -rf tmp
 }
 
 login() {
-  until docker exec -it forks_chainlink chainlink admin login -f /run/secrets/apicredentials
-  do
-      sleep 0.1
-  done
+  docker exec -it forks_chainlink chainlink admin login -f /run/secrets/apicredentials
 }
 
 create_job() {
+  login
   ETH_LOG_JOB=`cat fixtures/eth_log_job.json`
-	# wait for CL to start
-	search_chainlink_logs 'Listening and serving HTTP on port 6688'
-	# add ethlog job
-	login
-	docker exec -it forks_chainlink chainlink jobs create "$ETH_LOG_JOB"
+  docker exec -it forks_chainlink chainlink jobs create "$ETH_LOG_JOB"
 }
 
 create_contract() {
   CONTRACT_DATA=`cat fixtures/create_contract.json`
-	`curl \
-		-X POST \
-		-H "Content-Type: application/json" \
-		--data "$CONTRACT_DATA" \
-		172.16.1.100:8545`
+  curl \
+    -X POST \
+    -H "Content-Type: application/json" \
+    --data "$CONTRACT_DATA" \
+    172.16.1.100:8545
 }
 
 initial_setup() {
   tear_down
-	build_images
+  build_images
   start_network
+  waitForResponse "172.16.1.102:6688"
   create_job
 }
 
 
 assert_not_in_chainlink_logs() {
-	echo "asserting \"$1\" not present ..."
-	num_found=`docker-compose logs chainlink | grep -c "$1"`
-	if [ $num_found != "0" ]; then
-		printf "\n"
-		echo "TEST FAILED - Found $1 in chainlink logs and shouldn't have"
-		exit 1
-	fi;
+  echo "asserting \"$1\" not present ..."
```

Comment:
Since this is asserting on an asynchronous process, should it be doing this repeatedly for some time? If chainlink is slow and hasn't yet printed anything to the logs, this will immediately return true. And then later chainlink could emit that actual log.

Reply:
We could make this asynchronous, although in the two places we use this function, we've already shut down chainlink and just want to make sure that a certain log was not emitted while the node was running. See [forks/test#27](https://github.com/smartcontractkit/chainlink/blob/develop/integration/forks/test#L27) and [forks/test#45](https://github.com/smartcontractkit/chainlink/blob/develop/integration/forks/test#L45).

Reply:
Ahh, that's cool then 👍